cmd/metal-api/internal/metal/image.go

```diff
@@ -3,15 +3,61 @@ package metal
 import (
 	"fmt"
 	"strings"
+	"time"
 )
 
 // An Image describes an image which could be used for provisioning.
 type Image struct {
 	Base
 	URL      string                    `rethinkdb:"url" json:"url"`
 	Features map[ImageFeatureType]bool `rethinkdb:"features" json:"features"`
+	// OS is parsed from id and is the first part, specifies operating system derivate, internal usage only
+	OS string `rethinkdb:"os" json:"os"`
+	// Version is parsed from id and is the second part, specifies operating system version, internal usage only
+	Version string `rethinkdb:"version" json:"version"`
+	// ExpirationDate defines the time in the future, when this image is not considered for machine allocations anymore
+	ExpirationDate time.Time `rethinkdb:"expirationDate" json:"expirationDate"`
+	// Classification defines the state of a version (preview, supported, deprecated)
+	// only informational, no action depending on the classification done
+	Classification VersionClassification `rethinkdb:"classification" json:"classification"`
 }
 
+// VersionClassification is the logical state of a version according to
+// https://github.wdf.sap.corp/kubernetes/kube-docs/wiki/Versioning-Policy
+type VersionClassification string
+
+const (
+	// ClassificationPreview indicates that a version has recently been added and not promoted to "Supported" yet.
+	// ClassificationPreview versions will not be considered for automatic OperatingSystem patch version updates.
+	ClassificationPreview VersionClassification = "preview"
+	// ClassificationSupported indicates that a patch version is the default version for the particular minor version.
+	// There is always exactly one supported OperatingSystem patch version for every still maintained OperatingSystem minor version.
+	// Supported versions are eligible for the automated OperatingSystem patch version update machines.
+	ClassificationSupported VersionClassification = "supported"
+	// ClassificationDeprecated indicates that a patch version should not be used anymore, should be updated to a new version
+	// and will eventually expire.
+	// Every version that is neither in preview nor supported is deprecated.
+	// All patch versions of not supported minor versions are deprecated.
+	ClassificationDeprecated VersionClassification = "deprecated"
+)
+
+var versionClassifications = map[string]VersionClassification{
+	"preview":    ClassificationPreview,
```

Comment:
if the strings are always equal you could do 
~~~
string(ClassificationPreview): ClassificationPreview
~~~
but this is ok too.

Reply:
good point, done